### PR TITLE
19 integrate ff structurecheck in ff run so we know for certain no issues arise due to bad folder stucture

### PR DESCRIPTION
**Pull request by jduijvenbode-WWF**
<!--

Addition of ff_structurecheck to ff_run

### Description

I added the option to give an error on incorrect checks that are critical and silent passes on correct folder formats. I created unit tests for the errors so that they actually give an error. I also fixed some minor bugs in ff_run (outdated lines, no real bugs)

### Requirements

not too sleepy

### Related Issues

#28 #19

### CODE REVIEW

**Review by ZillahCalle — CHANGES_REQUESTED** [review]

File: R/ff_run.R [hunk 1]
```diff
@@ -148,8 +145,9 @@ ff_run <- function(shape = NULL, country = NULL, prediction_dates = NULL,
     # ff prep to determine the sample size
     if (autoscale_sample && hasvalue(fltr_condition)) {
       if (verbose) {
```
[the remark this answers is not in the document]
Comment (jduijvenbode-WWF) [reply R/ff_run.R:147]: goddamn right i should

File: R/ff_run.R [hunk 2]
```diff
@@ -148,8 +145,9 @@ ff_run <- function(shape = NULL, country = NULL, prediction_dates = NULL,
...
-        ff_cat("Finding optimal sample size based on filter condition\n", color = "green")
+        ff_cat("Finding optimal sample size based on filter condition", color = "green")
       }
+      prep_folder <- file.path(ff_folder,"preprocessed")
       ff_prep_params_original <- list(
         datafolder = prep_folder, shape = shape, dates = train_dates,
         fltr_condition = fltr_condition, fltr_features = fltr_features,
```
[the remark this answers is not in the document]
Comment (jduijvenbode-WWF) [reply R/ff_run.R:153]: excellent stuff

File: R/ff_structurecheck.R [hunk 3]
```diff
@@ -24,128 +31,277 @@ ff_structurecheck <- function(shape, folder_path, check_date = NULL) {
...
+  }
+}
+
+check_tile_files <- function(folder, tile, subfolder, check_date, silent_on_pass) {
+  files <- list.files(file.path(folder, tile), pattern = "\\.tif$")
+  if (length(files) == 0) {
+    print_result("No tif files in", subfolder, "for tile", tile,
+      color = "red", error_on_issue = TRUE
+    )
+    return()
```
[the remark this answers is not in the document]
Comment (jduijvenbode-WWF) [reply R/ff_structurecheck.R:72]: you're right. My code is dumb and i'm dumb

File: R/ff_structurecheck.R [hunk 4]
```diff
@@ -24,128 +31,277 @@ ff_structurecheck <- function(shape, folder_path, check_date = NULL) {
...
+  }
+
+  print_result("Subfolder for group", group, "present in models",
+    color = "green",
+    silent_on_pass = silent_on_pass
+  )
+
+  check_model_files(group_folder, group, error_on_issue, silent_on_pass)
```
[the remark this answers is not in the document]
Comment (jduijvenbode-WWF) [reply R/ff_structurecheck.R:205]: this is nitpicking. i can't take this seriously

File: R/ff_structurecheck.R [hunk 5]
```diff
@@ -24,128 +31,277 @@ ff_structurecheck <- function(shape, folder_path, check_date = NULL) {
...
+    print_result(
+      "EWS features for check date", check_date, "present in", subfolder,
+      "for tile", tile,
+      color = "green",
+      silent_on_pass = silent_on_pass
+    )
+  }
+}
+
+check_groundtruth <- function(files, tile, check_date, silent_on_pass) {
```
[the remark this answers is not in the document]
Comment (jduijvenbode-WWF) [reply R/ff_structurecheck.R:129]: maybe? okay

**Pull request merged**